File: assignments/solutions/99-problems/clojure/99-problems.clj
; P01: Find the last element of a list.

(defn another-last [ xs ] (last xs))

; P02: Find the last but one element of a list.

(defn one-but-last [ xs ]
  (if (= 2 (count xs))
      (first xs)
      (recur (rest xs))))

; P03: Find the k-th element of a list.

(defn kth [ n xs ] (nth xs n))

; P04: Find the number of elements of a list.

(defn another-length [ xs ] (count xs))

; P05: Reverse a list.

(defn another-reverse [ xs ] (reverse xs))

File: assignments/solutions/99-problems/haskell/99-problems.hs
-- P01: Find the last element of a list.

last' xs = last xs

-- P02: Find the last but one element of a list.

lastButOne xs = last $ init xs

-- P03: Find the k-th element of a list.

kth k xs = xs !! k

-- P04: Find the number of elements of a list.

length' xs = length xs

-- P05: Reverse a list.

reverse' xs = reverse xs